Sprint 15

a-sheryshov opened the pull request:
feat: add-item-requests

a-sheryshov committed:
Sprint 15
feat: add-item-requests

a-sheryshov committed:
Sprint 15
fix: checkstyle

a-sheryshov committed:
Sprint 15
fix: test coverage

sergeymamonov commented on src/main/java/ru/practicum/shareit/user/service/UserServiceDbImpl.java:
```diff
@@ -8,9 +8,6 @@
-import javax.validation.constraints.Positive;
-
 @Service
 public class UserServiceDbImpl extends AbstractEntityServiceImpl<User, UserDto> implements UserService {
```
Зачем Db в названии класса? Это же сервис. Я бы не использовал.

a-sheryshov replied:
Думал оставить 2 реализации (InMemory и Db) - в итоге оставил одну. 
Fixed

sergeymamonov commented on src/test/java/ru/practicum/shareit/item/ItemControllerTest.java:
```diff
@@ -0,0 +1,136 @@
+    }
+
+    @Test
+    void getAllTest() throws Exception {
```
В тестах можно проще, не писать throws Exception, а использовать аннотацию @SneakyThrows

sergeymamonov reviewed (changes requested):
BookingShortDto - не очень понимаю, где вызывается метод isValidEndTime. Аналогично в классе Booking.

a-sheryshov committed:
Sprint 15
fix: @SneakyThrows vise throw Exception, class naming
feat: custom validation test
